iso8601DateFromat is deprecated

```
src/Amazonka/Data/Time.hs:82:14: warning: [-Wdeprecations]
     In the use of ‘iso8601DateFormat’
     (imported from Data.Time.Format, but defined in time-1.11.1.1-TD80tev7xWHqZrSJnyyvU:Data.Time.Format.Locale):
     Deprecated: "use "Data.Time.Format.ISO8601" functions instead"
    |
 82 |   format _ = iso8601DateFormat (Just "%XZ")
    |
```

--- a/lib/amazonka-core/amazonka-core.cabal
+++ b/lib/amazonka-core/amazonka-core.cabal
@@ -115,7 +115,6 @@ library
     , http-types            >=0.12
     , lens                  >=4.14
     , memory                >=0.6
-    , mtl                   >=2.2      && <2.4
     , regex-posix           >=0.96
     , resourcet             >=1.1
     , scientific            >=0.3

--- a/lib/amazonka-core/amazonka-core.cabal
+++ b/lib/amazonka-core/amazonka-core.cabal
@@ -156,7 +156,6 @@ test-suite tests
     , data-ordlist
     , http-conduit
     , http-types
-    , lens
     , QuickCheck
     , quickcheck-unicode
     , tasty

--- a/lib/amazonka-core/src/Amazonka/Data/Body.hs
+++ b/lib/amazonka-core/src/Amazonka/Data/Body.hs
@@ -46,7 +46,7 @@ getFileSize path = liftIO (IO.withBinaryFile path IO.ReadMode IO.hFileSize)
 -- @newtype@ for show/orhpan instance purposes.
 newtype ResponseBody = ResponseBody
   {body :: ConduitM () ByteString (ResourceT IO) ()}
-  deriving stock Generic
+  deriving stock (Generic)
 
 instance Show ResponseBody where
   show = const "ResponseBody { ConduitM () ByteString (ResourceT IO) () }"

--- a/lib/amazonka-core/src/Amazonka/Data/Path.hs
+++ b/lib/amazonka-core/src/Amazonka/Data/Path.hs
@@ -49,7 +49,7 @@ rawPath = Raw . strip . BS8.split sep . toPath
 data Encoding = NoEncoding | Percent
   deriving stock (Eq, Show)
 
-data Path :: Encoding -> * where
+data Path :: Encoding -> Type where
   Raw :: [ByteString] -> Path 'NoEncoding
   Encoded :: [ByteString] -> Path 'Percent
 

--- a/lib/amazonka-core/src/Amazonka/Data/Time.hs
+++ b/lib/amazonka-core/src/Amazonka/Data/Time.hs
@@ -123,7 +123,7 @@ parseFormattedTime = do
     <|> parse (format (Proxy @BasicTime))
     <|> parse (format (Proxy @AWSTime))
     -- Deprecated ISO8601 format exhibited in the AWS-supplied examples.
-    <|> parse (iso8601DateFormat (Just "%X%Q%Z"))
+    <|> parse "%FT%X%Q%Z"
     -- Exhaustive Failure
     <|> fail ("Failure parsing Time from value: " ++ show s)
 

--- a/lib/amazonka-core/src/Amazonka/Types.hs
+++ b/lib/amazonka-core/src/Amazonka/Types.hs
@@ -669,7 +669,7 @@ requestUnsigned Request {service = Service {..}, ..} r =
 -- | Specify how a request can be de/serialised.
 class AWSRequest a where
   -- | The successful, expected response associated with a request.
-  type AWSResponse a :: *
+  type AWSResponse a :: Type
 
   request ::
     -- | Overrides applied to the default 'Service'.

--- a/lib/amazonka/src/Amazonka/Env.hs
+++ b/lib/amazonka/src/Amazonka/Env.hs
@@ -200,7 +200,6 @@ configureService s = overrideService f
 globalTimeout :: Seconds -> Env' withAuth -> Env' withAuth
 globalTimeout n = overrideService $ \s -> s {Service.timeout = Just n}
 
-
 -- | Disable any retry logic for an 'Env', so that any requests will
 -- at most be sent once.
 once :: Env' withAuth -> Env' withAuth

